pydocx/export/html.py

```diff
@@ -361,6 +361,114 @@ def export_paragraph_property_indentation(self, paragraph, results):
 
         return results
 
+    def get_previous_level_paragraph(self, num_id, level_id):
+        level_id = int(level_id)
+
+        while True:
+            if level_id == 0:
+                prev_level_id = level_id
+            else:
+                prev_level_id = level_id - 1
+
+            prev_level_paragraphs = self.numbering_level_listing_track[num_id][prev_level_id]
+            if prev_level_paragraphs:
+                return prev_level_paragraphs[-1]
+
+            if prev_level_id == 0 and not prev_level_paragraphs:
+                # this is an edge case with older version of word when it may contain a sublist
+                # into a separate num_id.
+                break
+
+            level_id -= 1
+
+        return None
+
+    def export_listing_paragraph_property_indentation(self, paragraph, level_properties,
+                                                      include_text_indent=False):
+        style = {}
+
+        if not level_properties or not paragraph.has_numbering_properties:
+            return style
+
+        default_level_indentation = paragraph.get_numbering_default_level_indentation()
+
+        paragraph_properties = paragraph.properties
+
+        level_id = int(paragraph_properties.numbering_properties.level_id)
+        num_id = paragraph_properties.numbering_properties.num_id
+
+        level_ind_left = level_properties.to_int('indentation_left', default=0)
+        level_ind_hanging = level_properties.to_int('indentation_hanging', default=0)
+
+        paragraph_ind_left = paragraph_properties.to_int('indentation_left', default=0)
+        paragraph_ind_hanging = paragraph_properties.to_int('indentation_hanging', default=0)
+        paragraph_ind_first_line = paragraph_properties.to_int('indentation_first_line',
+                                                               default=0)
+
+        left = paragraph_ind_left or level_ind_left
+        hanging = paragraph_ind_hanging or level_ind_hanging
+
+        # at this point we have no info about indentation, so we keep the default one
+        if not left and not hanging:
```

Comment:
Well, if paragraph is part of the listing then it's level will contain the `left` or `hanging`. I don't think we can have a listing paragraph where we don't have `left/hanging` on it's level and on paragraph but first line is set.